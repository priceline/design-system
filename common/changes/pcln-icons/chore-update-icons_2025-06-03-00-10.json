{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}