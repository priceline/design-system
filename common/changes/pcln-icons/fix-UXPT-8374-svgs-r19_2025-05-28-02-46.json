{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}