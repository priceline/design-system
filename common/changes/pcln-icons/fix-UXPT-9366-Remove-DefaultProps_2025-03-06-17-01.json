{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}