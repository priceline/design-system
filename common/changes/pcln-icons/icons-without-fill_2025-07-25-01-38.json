{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}