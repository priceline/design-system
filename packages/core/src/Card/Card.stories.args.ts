import { borderStyles, colors, inputArgs, shadows } from '../storybook/args'

export const argTypes: unknown = {
  as: {
    control: 'select',
    description:
      'Alter the base render component. Here we can have the Card be rendered on the DOM as a `button` for accessibility. It grants the ability to tab and focus the element',
    options: ['', 'button'],
  },
  bg: {
    control: 'select',
    options: colors,
  },
  borderColor: {
    control: 'select',
    options: colors,
    defaultValue: 'border',
  },
  borderRadius: {
    control: 'select',
    options: inputArgs,
    defaultValue: 'xsm',
  },
  borderStyle: {
    control: 'select',
    options: borderStyles,
    defaultValue: 'solid',
  },
  borderWidth: {
    control: { type: 'number', min: 0, max: 5, step: 1 },
    defaultValue: 1,
  },
  boxShadowSize: {
    control: 'select',
    options: shadows,
    defaultValue: 'md',
  },
}
