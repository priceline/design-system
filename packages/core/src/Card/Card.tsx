import React from 'react'
import styled from 'styled-components'
import { border, BorderProps } from 'styled-system'
import { Box, type BoxProps } from '../Box/Box'
import { applyVariations, getPaletteColor } from '../utils/utils'

const styleAsButton = ({ as, ...props }) =>
  as === 'button'
    ? `
      font-family : inherit;
      &:hover {
        cursor: pointer;
        box-shadow: ${props.theme.shadows.xl};
      }`
    : ''

/**
 * @public
 */
export type CardProps = BoxProps &
  BorderProps & {
    borderColor?: string
  }

/**
 * @public
 */
export const Card: React.FC<CardProps> = styled(Box)`
  ${applyVariations('Card')}
  ${styleAsButton}

  border-color: ${(props) =>
    props.borderWidth === 0 ? '0' : getPaletteColor(props.borderColor, 'base')(props)};
  ${border};
`

Card.defaultProps = {
  borderColor: 'border',
  borderRadius: 'xsm',
  borderStyle: 'solid',
  borderWidth: 1,
}

Card.displayName = 'Card'
