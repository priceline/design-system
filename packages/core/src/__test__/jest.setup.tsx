import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
