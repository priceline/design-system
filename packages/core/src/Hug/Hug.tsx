import React, { Attributes } from 'react'
import { type CardProps } from '../Card/Card'
import { Text } from '../Text/Text'
import { BorderConcealer, Header, HugCard } from './Hug.styled'

/**
 * @public
 */
export type HugProps = CardProps & {
  children?: React.ReactNode
  iconDisplay?: string[]
  icon?: React.ReactNode
  text?: React.ReactNode
  color?: string
  fontSize?: string | number
}

/**
 * @public
 */
export function Hug({
  bg,
  borderRadius = 'xl',
  borderWidth = 1,
  color = 'text.lightest',
  p = 2,
  fontSize = 1,
  icon,
  iconDisplay,
  colorScheme,
  ...props
}: HugProps): React.ReactElement {
  let iconClone
  if (React.isValidElement(icon)) {
    iconClone = React.cloneElement(icon, {
      style: { display: iconDisplay },
      mr: 2,
      mt: '-2px',
      mb: '2px',
      size: 24,
    } as unknown as Attributes)
  }

  let headerColor = ''

  if (colorScheme && color !== 'text.lightest') {
    headerColor = color
  } else if (!colorScheme) {
    headerColor = color
  }

  return (
    <HugCard
      {...props}
      borderColor={colorScheme ? null : bg || color}
      borderRadius={borderRadius}
      borderWidth={borderWidth}
      color={color}
      colorScheme={colorScheme}
    >
      <Header
        bg={bg}
        color={headerColor}
        p={p}
        pl='12px'
        alignItems='center'
        colorScheme={colorScheme}
        iconUsesColorScheme={!iconClone?.props?.color}
      >
        {!!iconClone && iconClone}
        <Text fontSize={fontSize}>{props.text}</Text>
      </Header>
      <BorderConcealer>{props.children}</BorderConcealer>
    </HugCard>
  )
}
