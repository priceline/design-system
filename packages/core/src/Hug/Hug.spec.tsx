import { ThumbsUp } from 'pcln-icons'
import { Card } from '../Card/Card'
import { Text } from '../Text/Text'
import { render } from '../__test__/testing-library'
import { Hug } from './Hug'

const text = (
  <Text.span>
    Have you seen this <Text.span bold>card</Text.span> I&ampls;m hugging?!
  </Text.span>
)

describe('Hug', () => {
  test('renders text, icon, and Child', () => {
    const { asFragment } = render(
      <Hug icon={<ThumbsUp />} text={text}>
        <Card p={3} bg='white' color='text'>
          I&apos;m a card within a hug!
        </Card>
      </Hug>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders text when string is provided', () => {
    const { asFragment } = render(
      <Hug icon={<ThumbsUp />} text="I've been known to hug a card from time to time">
        <Card p={3} bg='white' color='text'>
          I&apos;m a card within a hug!
        </Card>
      </Hug>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders text when array of nodes is provided', () => {
    const { asFragment } = render(
      <Hug icon={<ThumbsUp />} text={[text, text]}>
        <Card p={3} bg='white' color='text'>
          I&apos;m a card within a hug!
        </Card>
      </Hug>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
