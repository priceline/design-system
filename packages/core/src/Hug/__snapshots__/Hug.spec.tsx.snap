// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hug renders text when array of nodes is provided 1`] = `
<DocumentFragment>
  .c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-bottom: 2px;
  margin-right: 8px;
  width: 24px;
}

.c7 {
  outline: none;
}

.c1 {
  background-color: #fff;
  color: #001833;
  border-radius: 16px;
}

.c4 {
  background-color: #fff;
  color: #001833;
  padding: 8px;
  padding-left: 12px;
}

.c11 {
  background-color: #fff;
  color: #001833;
  border-radius: 2px;
  padding: 16px;
}

.c2 {
  border-color: #fff;
  border-color: text.lightest;
  border-radius: 16px;
  border-style: solid;
  border-width: 1px;
}

.c12 {
  border-color: #c0cad5;
  border-color: #c0cad5;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
}

.c8 {
  font-size: 14px;
}

.c9 {
  font-weight: 700;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  overflow: hidden;
}

.c10 > * {
  border: 0 !important;
}

<div
    class="c0"
  >
    <div
      class="c1 c2 c3"
      color="text.lightest"
    >
      <div
        class="c4 c5 "
        color="text.lightest"
      >
        <svg
          aria-hidden="true"
          class="c6 c7"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          size="24"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 20h3.3v-9.6H3V20zm18-8.8c0-.9-.7-1.6-1.6-1.6h-5.2l.8-3.7v-.2c0-.3-.2-.6-.3-.9l-.9-.8-5.4 5.3c-.3.2-.5.6-.5 1.1v8c0 .9.7 1.6 1.6 1.6h7.4c.7 0 1.2-.4 1.5-1l2.5-5.7c.1-.1.1-.3.1-.5v-1.6z"
          />
        </svg>
        <div
          class="c8"
          font-size="1"
        >
          <span
            class=""
          >
            Have you seen this 
            <span
              class="c9"
            >
              card
            </span>
             I&ampls;m hugging?!
          </span>
          <span
            class=""
          >
            Have you seen this 
            <span
              class="c9"
            >
              card
            </span>
             I&ampls;m hugging?!
          </span>
        </div>
      </div>
      <div
        class="c10"
      >
        <div
          class="c11 c12"
          color="text"
        >
          I'm a card within a hug!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hug renders text when string is provided 1`] = `
<DocumentFragment>
  .c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-bottom: 2px;
  margin-right: 8px;
  width: 24px;
}

.c7 {
  outline: none;
}

.c1 {
  background-color: #fff;
  color: #001833;
  border-radius: 16px;
}

.c4 {
  background-color: #fff;
  color: #001833;
  padding: 8px;
  padding-left: 12px;
}

.c10 {
  background-color: #fff;
  color: #001833;
  border-radius: 2px;
  padding: 16px;
}

.c2 {
  border-color: #fff;
  border-color: text.lightest;
  border-radius: 16px;
  border-style: solid;
  border-width: 1px;
}

.c11 {
  border-color: #c0cad5;
  border-color: #c0cad5;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
}

.c8 {
  font-size: 14px;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  overflow: hidden;
}

.c9 > * {
  border: 0 !important;
}

<div
    class="c0"
  >
    <div
      class="c1 c2 c3"
      color="text.lightest"
    >
      <div
        class="c4 c5 "
        color="text.lightest"
      >
        <svg
          aria-hidden="true"
          class="c6 c7"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          size="24"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 20h3.3v-9.6H3V20zm18-8.8c0-.9-.7-1.6-1.6-1.6h-5.2l.8-3.7v-.2c0-.3-.2-.6-.3-.9l-.9-.8-5.4 5.3c-.3.2-.5.6-.5 1.1v8c0 .9.7 1.6 1.6 1.6h7.4c.7 0 1.2-.4 1.5-1l2.5-5.7c.1-.1.1-.3.1-.5v-1.6z"
          />
        </svg>
        <div
          class="c8"
          font-size="1"
        >
          I've been known to hug a card from time to time
        </div>
      </div>
      <div
        class="c9"
      >
        <div
          class="c10 c11"
          color="text"
        >
          I'm a card within a hug!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hug renders text, icon, and Child 1`] = `
<DocumentFragment>
  .c6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-bottom: 2px;
  margin-right: 8px;
  width: 24px;
}

.c7 {
  outline: none;
}

.c1 {
  background-color: #fff;
  color: #001833;
  border-radius: 16px;
}

.c4 {
  background-color: #fff;
  color: #001833;
  padding: 8px;
  padding-left: 12px;
}

.c11 {
  background-color: #fff;
  color: #001833;
  border-radius: 2px;
  padding: 16px;
}

.c2 {
  border-color: #fff;
  border-color: text.lightest;
  border-radius: 16px;
  border-style: solid;
  border-width: 1px;
}

.c12 {
  border-color: #c0cad5;
  border-color: #c0cad5;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
}

.c8 {
  font-size: 14px;
}

.c9 {
  font-weight: 700;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  overflow: hidden;
}

.c10 > * {
  border: 0 !important;
}

<div
    class="c0"
  >
    <div
      class="c1 c2 c3"
      color="text.lightest"
    >
      <div
        class="c4 c5 "
        color="text.lightest"
      >
        <svg
          aria-hidden="true"
          class="c6 c7"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          size="24"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 20h3.3v-9.6H3V20zm18-8.8c0-.9-.7-1.6-1.6-1.6h-5.2l.8-3.7v-.2c0-.3-.2-.6-.3-.9l-.9-.8-5.4 5.3c-.3.2-.5.6-.5 1.1v8c0 .9.7 1.6 1.6 1.6h7.4c.7 0 1.2-.4 1.5-1l2.5-5.7c.1-.1.1-.3.1-.5v-1.6z"
          />
        </svg>
        <div
          class="c8"
          font-size="1"
        >
          <span
            class=""
          >
            Have you seen this 
            <span
              class="c9"
            >
              card
            </span>
             I&ampls;m hugging?!
          </span>
        </div>
      </div>
      <div
        class="c10"
      >
        <div
          class="c11 c12"
          color="text"
        >
          I'm a card within a hug!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
